sklearn/manifold/tests/test_mds.py

```diff
@@ -52,10 +52,98 @@ def test_smacof_error():
     assert_raises(ValueError, mds.smacof, sim, init=Z, n_init=1)
 
 
-def test_MDS():
+def test_fit_error():
+    # fit() should throw an error if the dissimilarity metric is unknown
+    sim = np.array([[0, 5, 3, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    mds_clf = mds.MDS(dissimilarity="unknown")
+    assert_raises(ValueError, mds_clf.fit, sim)
+
+    mds_clf = mds.MDS(dissimilarity="unknown", extendible=True)
+    assert_raises(ValueError, mds_clf.fit, sim)
+
+
+def test_fit_transform_error():
+    sim = np.array([[0, 5, 3, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    Z = np.array([[-.266, -.539],
+                  [.451, .252],
+                  [.016, -.238],
+                  [-.200, .524]])
+
+    # fit_transform should throw an error if init is not None
+    # and dissimilarity="euclidean"
+    mds_clf = mds.MDS(dissimilarity="euclidean", extendible=True)
+    assert_raises(ValueError, mds_clf.fit_transform, sim, init=Z)
+
+
+def test_MDS_fit_transform():
     sim = np.array([[0, 5, 3, 4],
                     [5, 0, 2, 2],
                     [3, 2, 0, 1],
                     [4, 2, 1, 0]])
     mds_clf = mds.MDS(metric=False, n_jobs=3, dissimilarity="precomputed")
     mds_clf.fit(sim)
+    mds_clf.fit_transform(sim)
+
+    # Test non-parallel case
+    mds_clf = mds.MDS(metric=False, n_jobs=1, dissimilarity="precomputed")
+    mds_clf.fit(sim)
+    mds_clf.fit_transform(sim)
+
+
+def test_MDS_fit_transform_extend():
+    sim = np.array([[0, 5, 3, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    Z = np.array([[-.266, -.539],
+                  [.451, .252],
+                  [.016, -.238],
+                  [-.200, .524]])
+
+    # test fit_transform under the extendible case
+    mds_clf = mds.MDS(dissimilarity="euclidean", extendible=True)
+    mds_clf.fit_transform(sim)
```

Comment:
Should the output of this be identical to the extendible=False case?

Reply:
I don't believe it should be the same.

Reply:
It would not, the inductive=False case uses the SMACOF algorithm to find projection coordinates for the points, which is an iterative minimisation procedure on a loss function ("stress"). The inductive=True case produces the closed form solution of this minimisation, taking the loss function to be the "strain" (special case of the general "stress"). The closed form solution involves performing eigendecomposition on the centred dissimilarity matrix.